R/dependecies.R

```diff
@@ -0,0 +1,510 @@
+  }
+  if (!is.na(locked_package$version) && exists_in_lockbox(locked_package)) {
+    dependencies <- dependencies_from_description(locked_package
+      , description_file_for(locked_package$name
+        , gsub("/[^/]+$", "", lockbox_package_path(locked_package))))
+  } else {
+    cat(crayon::blue("."))
+    output <- tryCatch(get_remote_dependencies(package), error = function(e) e)
+    if (is(output, "error")) {
+      message(crayon::red(paste0("Dependencies could not be resolved for package: "
+        , package$name, " version: ", package$version)))
+      dependencies <- list()
+    } else {
+      dependencies <- output
+    }
+  }
+  dependencies <- strip_available_dependencies(dependencies)
+  dependencies <- lapply(dependencies, add_latest_version_in_lockbox)
+  lapply(dependencies, replace_with_lock, lock)
+}
+
+strip_available_dependencies <- function(dependencies) {
+  dependencies <- lapply(dependencies
+    , function(package) {
+      if(package$name %in% pesky_namespaces) NULL
+      else package
+    })
+ dependencies[!vapply(dependencies, is.null, logical(1))]
+}
+
+#' Get the dependencies for a given package
+get_remote_dependencies <- function(package) {
+  UseMethod("get_remote_dependencies")
+}
+
+#' If a package is local we just read from the directory given
+get_remote_dependencies.local <- function(package) {
+  description_name <- file_list$Name[grepl(paste0("^[^/]+"
+    ,"/DESCRIPTION$"), file_list$Name)]
+  dcf <- read.dcf(description_name)
+  dependencies_from_description(package, dcf)
+}
+
+#' For packages on CRAN we will extract to a temporary directory when we
+#' download the accurate remote DESCRIPTION file.
+get_remote_dependencies.CRAN <- function(package) {
+  if (package$is_dependency_package) {
+    package$version <- NA
+  }
+  filepath <- download_package(package)
+  split_fp <- strsplit(filepath, "/")[[1]]
+  dirpath <- paste(split_fp[-length(split_fp)], collapse = "/")
+  file_list <- untar(filepath, list = TRUE)
+  description_name <- file_list[grepl(paste0("^[^/]+"
+    ,"/DESCRIPTION$"), file_list)]
+  output <- untar(filepath, description_name, exdir = dirpath)
+  description_path <- paste0(dirpath, "/", description_name)
+  dcf <- read.dcf(file = description_path)
+  unlink(filepath)
+  dependencies_from_description(package, dcf)
+}
+
+#' Download the accurate remote DESCRIPTION file for a github repo.
+get_remote_dependencies.github <- function(package) {
+  dcf <- download_description_github(package)
+  dependencies_from_description(package, dcf)
+}
+
+download_description_github <- function(package) {
+  remote <- package$remote
+  filepath <- download_package(structure(
+    package,
+    class = c(remote, class(package))))
+  file_list <- unzip(filepath, list = TRUE)
+  subdir <- ""
+  if (!is.null(package$subdir)){
+    subdir <- paste0("/",package$subdir)
+  }
+  description_name <- file_list$Name[grepl(paste0("^[^/]+"
+    , subdir
+    ,"/DESCRIPTION$"), file_list$Name)]
```

Comment:
Instead of:

```R
  subdir <- ""
  if (!is.null(package$subdir)){
    subdir <- paste0("/",package$subdir)
  }
  description_name <- file_list$Name[grepl(paste0("^[^/]+"
    , subdir
    ,"/DESCRIPTION$"), file_list$Name)]
```

I think you can just do:

```R
description_name <- grep(file.path(package$subdir, "DESCRIPTION"), file_list$Name, fixed = TRUE, value = TRUE)
```